src/utils/image_utils.py

```
@@ -230,3 +230,115 @@ def display_image(image: np.ndarray, name="image", scale=None):
+    vertical_min = max(4, int(zone_h * 0.5))
+    for contour in yellow_contours:
+        x, y, w, h = cv2.boundingRect(contour)
+        if h >= vertical_min and w <= 30:
+            pointer_candidates.append((x, y, w, h))
+    if pointer_candidates:
+        pointer_x, pointer_y, pointer_w, pointer_h = max(
+            pointer_candidates, key=lambda item: item[3] * max(1, item[2])
+        )
+        pointer_center = pointer_x + pointer_w // 2
+    else:
+        # 兜底：按黄线列投影找最亮竖线（避免轮廓断裂时丢指针）
```

Ruff (0.15.11)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)